### PR TITLE
Private/mmeeks/clipboardfix

### PR DESCRIPTION
**Pull request by mmeeks**
* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: browser/src/map/Clipboard.js [hunk 1]
```diff
@@ -828,26 +851,44 @@ L.Clipboard = L.Class.extend({
...
+			try {
+				var result = fetch(url).then(response => response.text());
+				resolve(result);
+			} catch (err) {
+				reject(err);
+			}
+		});
+
+		var awaitPromise = function(url, mimetype, shorttype) {
+			return new Promise((resolve, reject) => {
```
Comment (vmiklos) [inline browser/src/map/Clipboard.js:881]: Oh nice, I didn't consider that the callback can be shared to fetch the data only once.

File: wsd/protocol.txt [hunk 2]
```diff
@@ -242,9 +242,12 @@ dialog <command>
 
     <command> is unique identifier for the dialog that needs to be painted.
 
-uno <command>
+uno <command> [serial=text] [arguments]
```
Comment (vmiklos) [inline wsd/protocol.txt:245]: Is this 'serial=' correct? I don't see any output from 'git grep serial=', but perhaps I miss something.

**Review by vmiklos — APPROVED** [review]
The actual race fix looks good to me, thanks for noticing this. I'm not yet merging, possibly you want to add more commits to this PR first. I also didn't check if the CI failure is unrelated or not.

File: browser/src/map/Clipboard.js (hunk 1 above)
Comment (mmeeks) [reply browser/src/map/Clipboard.js:881]: =)

Comment (mmeeks) [reply browser/src/map/Clipboard.js:881]: Ah yes, this was my 1st thought at adding a new reply with a serial in it so we can undo any races related to which UNO command was invoked and which the reply is for - which would be by far more ideal, but is a bit vicious to get threaded through the core & API & so on ;-)

**Comment by mmeeks** [comment]
CI failure is clearly un-related - this code-change can't affect C++ unit tests =)

**Comment by vmiklos** [comment]
https://cpci.cbg.collabora.co.uk:8080/job/github_online_master_debug_vs_co-24.04/1450/console fails in unit-integration, let's see if this is a one-off, retry.

**Pull request merged**